src/components/GameGrid.jsx

```diff
@@ -1,9 +1,8 @@
 import GridBox from './GridBox'
 
-export default function GameGrid({ cells }) {
-
+export default function GameGrid({ cells, gameOverScreen }) {
     return (
-        <div className="grid">
+        <div className={"grid" + ((gameOverScreen) ? " blur" : "")}>
```

Comment:
I don't have much experience with React but this seems like a naive way of doing. Surely React will have some meaningful way like a variable whose value is class name and you can update that variable in the code. This would separate your logic code from UI code.

Reply:
extracted classname to a variable